README.md

```diff
@@ -1,111 +1,61 @@
-int redisClusterSetOptionAddNodes(redisClusterContext *cc, const char *addrs);
-int redisClusterSetOptionConnectBlock(redisClusterContext *cc);
-int redisClusterSetOptionConnectNonBlock(redisClusterContext *cc);
-int redisClusterSetOptionParseSlaves(redisClusterContext *cc);
-int redisClusterSetOptionParseOpenSlots(redisClusterContext *cc);
-int redisClusterSetOptionRouteUseSlots(redisClusterContext *cc);
-int redisClusterSetOptionConnectTimeout(redisClusterContext *cc, const struct timeval tv);
-int redisClusterSetOptionTimeout(redisClusterContext *cc, const struct timeval tv);
-int redisClusterSetOptionMaxRedirect(redisClusterContext *cc,  int max_redirect_count);
+* SSL/TLS
+    * Connect to Redis nodes using SSL/TLS (supported from Redis 6)
 
-int redisClusterConnect2(redisClusterContext *cc);
+* IPv6
+    * Handles clusters on IPv6 networks
 
-void *redisClusterFormattedCommand(redisClusterContext *cc, char *cmd, int len);
-void *redisClustervCommand(redisClusterContext *cc, const char *format, va_list ap);
-void *redisClusterCommand(redisClusterContext *cc, const char *format, ...);
-void *redisClusterCommandArgv(redisClusterContext *cc, int argc, const char **argv, const size_t *argvlen);
-int redisClusterAppendFormattedCommand(redisClusterContext *cc, char *cmd, int len);
-int redisClustervAppendCommand(redisClusterContext *cc, const char *format, va_list ap);
-int redisClusterAppendCommand(redisClusterContext *cc, const char *format, ...);
-int redisClusterAppendCommandArgv(redisClusterContext *cc, int argc, const char **argv, const size_t *argvlen);
-int redisClusterGetReply(redisClusterContext *cc, void **reply);
-void redisClusterReset(redisClusterContext *cc);
-
-redisContext *ctx_get_by_node(redisClusterContext *cc, struct cluster_node *node);
-
-redisClusterAsyncContext *redisClusterAsyncConnect(const char *addrs, int flags);
-int redisClusterAsyncSetConnectCallback(redisClusterAsyncContext *acc, redisConnectCallback *fn);
-int redisClusterAsyncSetDisconnectCallback(redisClusterAsyncContext *acc, redisDisconnectCallback *fn);
-int redisClusterAsyncFormattedCommand(redisClusterAsyncContext *acc, redisClusterCallbackFn *fn, void *privdata, char *cmd, int len);
-int redisClustervAsyncCommand(redisClusterAsyncContext *acc, redisClusterCallbackFn *fn, void *privdata, const char *format, va_list ap);
-int redisClusterAsyncCommand(redisClusterAsyncContext *acc, redisClusterCallbackFn *fn, void *privdata, const char *format, ...);
-int redisClusterAsyncCommandArgv(redisClusterAsyncContext *acc, redisClusterCallbackFn *fn, void *privdata, int argc, const char **argv, const size_t *argvlen);
-
-void redisClusterAsyncDisconnect(redisClusterAsyncContext *acc);
-void redisClusterAsyncFree(redisClusterAsyncContext *acc);
+## Build instructions
 
-redisAsyncContext *actx_get_by_node(redisClusterAsyncContext *acc, cluster_node *node);
+```sh
+$ mkdir build; cd build
+$ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
```

Comment:
I think this example line should contain at least the DOWNLOAD_HIREDIS flag, so that it actually works to just run this and use the library. And possibly also the ENABLE_SSL flag, as an example.

Btw, how do you link with hiredis-cluster and hiredis? Do you need to link to both libraries or is hiredis included in hiredis-cluster?

Reply:
Updated, and better described some details